projection/channel/view/channels.go

```diff
@@ -0,0 +1,354 @@
+	if err != nil {
+		return fmt.Errorf("error updating total_transfer_out_success_rate: %v: %w", err, rdb.ErrWrite)
+	}
+	if result.RowsAffected() != 1 {
+		return fmt.Errorf("error updating total_transfer_out_success_rate: no row updated: %w", rdb.ErrWrite)
+	}
+
+	return nil
+}
+
+func (channelsView *Channels) UpdateLastActivityTime(channelID string, portID string, time utctime.UTCTime) error {
+	sql, sqlArgs, err := channelsView.rdb.StmtBuilder.
+		Update("view_ibc_channels").
+		SetMap(map[string]interface{}{
+			"last_activity_time": channelsView.rdb.Tton(&time),
+		}).
+		Where("channel_id = ? AND port_id = ?", channelID, portID).
+		ToSql()
+	if err != nil {
+		return fmt.Errorf("error building channel update sql: %v: %w", err, rdb.ErrBuildSQLStmt)
+	}
+
+	result, err := channelsView.rdb.Exec(sql, sqlArgs...)
+	if err != nil {
+		return fmt.Errorf("error updating last_activity_time: %v: %w", err, rdb.ErrWrite)
+	}
+	if result.RowsAffected() != 1 {
+		return fmt.Errorf("error updating last_activity_time: no row updated: %w", rdb.ErrWrite)
+	}
+
+	return nil
+}
+
+func (channelsView *Channels) List(order ChannelsListOrder, pagination *pagination.Pagination) ([]ChannelRow, *pagination.PaginationResult, error) {
+	stmtBuilder := channelsView.rdb.StmtBuilder.Select(
+		"channel_id",
+		"port_id",
+		"connection_id",
+		"counterparty_channel_id",
+		"counterparty_port_id",
+		"packet_ordering",
+		"last_in_packet_sequence",
+		"last_out_packet_sequence",
+		"total_transfer_in_count",
+		"total_transfer_out_count",
+		"total_transfer_out_success_count",
+		"total_transfer_out_success_rate",
+		"last_activity_time",
+		"bonded_tokens",
+	).From(
+		"view_ibc_channels",
+	)
+
+	if order.ChannelId == view.ORDER_DESC {
+		stmtBuilder = stmtBuilder.OrderBy("channel_id DESC")
+	} else {
+		stmtBuilder = stmtBuilder.OrderBy("channel_id")
+	}
+
+	rDbPagination := rdb.NewRDbPaginationBuilder(
+		pagination,
+		channelsView.rdb,
+	).BuildStmt(stmtBuilder)
+	sql, sqlArgs, err := rDbPagination.ToStmtBuilder().ToSql()
+	if err != nil {
+		return nil, nil, fmt.Errorf("error building channels select SQL: %v, %w", err, rdb.ErrBuildSQLStmt)
+	}
+
+	rowsResult, err := channelsView.rdb.Query(sql, sqlArgs...)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error executing channels select SQL: %v: %w", err, rdb.ErrQuery)
+	}
+	defer rowsResult.Close()
+
+	channels := make([]ChannelRow, 0)
+	for rowsResult.Next() {
+		var channel ChannelRow
+		var bondedTokensJSON string
+		if err = rowsResult.Scan(
+			&channel.ChannelID,
+			&channel.PortID,
+			&channel.ConnectionID,
+			&channel.CounterpartyChannelID,
+			&channel.CounterpartyPortID,
+			&channel.PacketOrdering,
+			&channel.LastInPacketSequence,
+			&channel.LastOutPacketSequence,
+			&channel.TotalTransferInCount,
+			&channel.TotalTransferOutCount,
+			&channel.TotalTransferOutSuccessCount,
+			&channel.TotalTransferOutSuccessRate,
+			&channel.LastActivityTime,
```

Comment:
Here you should use a native to time reader

Reply:
https://github.com/crypto-com/chain-indexing/blob/05a6ed0e619efc89ee88841021996d3e4ab3c56f/projection/block/view/blocks.go#L118

Reply:
Ahh sorry I missed it... Let me add the time reader!